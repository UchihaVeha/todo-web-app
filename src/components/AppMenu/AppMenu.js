// @flow
import * as React from 'react';
import { compose, pure } from 'recompose';
import { Link } from 'react-router-dom';
import { withStyles } from 'material-ui/styles';
import { Drawer, Button } from 'material-ui';
import List, { ListItem } from 'material-ui/List';

const styles = () => ({
  drawerPaper: {
    width: 250
  },
  button: {
    width: '100%',
    textTransform: 'none',
    justifyContent: 'flex-start'
  }
});

type Props = {
  classes: Return<typeof styles>,
  handleMenuToggle: () => void,
  open: boolean
};

export const menuItems = [{ title: 'Home', url: '/' }];

export const AppMenuComponent = ({
  open,
  handleMenuToggle,
  classes
}: Props) => (
  <Drawer
    open={open}
    onRequestClose={handleMenuToggle}
    ModalProps={{
      keepMounted: true // Better open performance on mobile.
    }}
    classes={{
      paper: classes.drawerPaper
    }}
  >
    <List>
      {menuItems.map(item => (
        <ListItem key={item.url} disableGutters>
          <Button
            disableRipple
            to={item.url}
            component={Link}
            className={classes.button}
          >
            {item.title}
          </Button>
        </ListItem>
      ))}
    </List>
  </Drawer>
);

export default compose(pure, withStyles(styles, { name: 'AppMenu' }))(
  AppMenuComponent
);
