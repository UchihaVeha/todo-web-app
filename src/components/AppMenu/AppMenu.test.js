import React from 'react';
import { shallow } from 'enzyme';
import AppMenu, { menuItems, AppMenuComponent } from './AppMenu';

describe('AppMenu component', () => {
  const wrapper = (
    { open = false, handleMenuToggle = jest.fn(), classes = {} } = {}
  ) =>
    shallow(
      <AppMenuComponent
        open={open}
        handleMenuToggle={handleMenuToggle}
        classes={classes}
      />
    );

  it('should render component', () => {
    expect(wrapper()).toMatchSnapshot();
  });

  it('should have enhance', () => {
    expect(shallow(<AppMenu />)).toMatchSnapshot();
  });

  it('should have menu items', () => {
    expect(menuItems).toMatchSnapshot();
  });
});
