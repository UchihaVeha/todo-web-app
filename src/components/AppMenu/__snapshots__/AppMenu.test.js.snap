// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMenu component should have enhance 1`] = `<withStyles(AppMenuComponent) />`;

exports[`AppMenu component should have menu items 1`] = `
Array [
  Object {
    "title": "Home",
    "url": "/",
  },
]
`;

exports[`AppMenu component should render component 1`] = `
<withStyles(Drawer)
  ModalProps={
    Object {
      "keepMounted": true,
    }
  }
  classes={
    Object {
      "paper": undefined,
    }
  }
  onRequestClose={[Function]}
  open={false}
>
  <withStyles(List)>
    <withStyles(ListItem)
      disableGutters={true}
      key="/"
    >
      <withStyles(Button)
        component={[Function]}
        disableRipple={true}
        to="/"
      >
        Home
      </withStyles(Button)>
    </withStyles(ListItem)>
  </withStyles(List)>
</withStyles(Drawer)>
`;
